### PR TITLE
Open Api 3 support and documentation

### PR DESCRIPTION
**Pull request by zekth**
As said in #727 here is the Open Api 3 support.

Spec file is currently in the process of writing.  The open api spec file is written in YAML and located in `build/openapi.yaml`

To generate the documentation we use the redoc-cli https://github.com/Rebilly/ReDoc which give us the flexibility to generate a no dependency html file to be served by the server. The generation of the doc is done by calling `npm run doc:api`

The documentation is reachable at this url `http://reef-pi-url/api-doc`

Documentation looks like this:
2019-02-27 10_41_06-redoc documentation

### CODE REVIEW

**Comment by ranjib** [comment]
👍

File: package.json [hunk 1]
```diff
@@ -69,13 +70,14 @@
     "yup": "^0.26.2"
   },
   "scripts": {
+    "doc:api":"./node_modules/.bin/redoc-cli bundle build/openapi.yaml -o bin/api-doc.html",
     "translations": "node ./build/translations/csv-json.js",
     "translations:sync": "node ./build/translations/csv-sync.js",
     "build": "npm run translations && ./node_modules/.bin/webpack -d",
     "ui": "npm run translations && ./node_modules/.bin/webpack --mode=production",
-    "ui-dev": "npm run translations && ./node_modules/.bin/webpack -d --watch",
-    "js-lint": "./node_modules/.bin/standard  front-end/src/**/*",
```
Comment (ranjib) [inline package.json]: @zekth this is correct?

Comment (zekth) [reply package.json]: js-lint was not used anywhere. But after seeing this. Deleting ui-dev is wrong. I'll fix this. Sorry

Comment (zekth) [reply package.json]: Totally wrong... Don't remember why i've done that. Will rollback sorry.

**Comment by ranjib** [comment]
maintaining this yaml file will be some work. I am wonderig if theres a go middleware that can generate this automatically

**Comment by zekth** [comment]
> maintaining this yaml file will be some work. I am wonderig if theres a go middleware that can generate this automatically

I've tried to find but there is no such implementation in golang for mux

**Comment by ranjib** [comment]
https://github.com/reef-pi/reef-pi/pull/839
@zekth planning to push this lightweight solution. We wont have security header etc, but it will give us pretty solid starting point which i can then just polish a bit to get full blown doc page that will go in the website

**Pull request closed without merging**